File: solutions/1878-check-if-array-is-sorted-and-rotated/solution.java
class Solution {
    public boolean check(int[] nums) {
        int start=-1,flag=0,n=nums.length;
        for(int i=0;i<n-1;i++)
        {
            if(flag==0)
                if(nums[i]>nums[i+1])
                {
                    start=i;
                    flag=1;
                }
            if(start!=i)
                if(nums[i]>nums[i+1])
                    return false;
        }
        if(start==-1)
            return true;
        if(nums[0]<nums[n-1])
            return false;
        return true;

    }
}

File: solutions/1878-check-if-array-is-sorted-and-rotated/solution.py
class Solution(object):
    def check(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        start=-1
        flag=0
        for i in range(len(nums)-1):
            if flag==0:
                if nums[i]>nums[i+1]:
                    start=i
                    flag=1
            if start!=i:
                if nums[i]>nums[i+1]:
                    return False
        if start==-1:
            return True
        if nums[0]<nums[len(nums)-1]:
            return False
        return True

